### PR TITLE
John4744/rtc fixes

### PR DESCRIPTION
**Pull request by john4744**
This PR is to apply the various fixes from our previous 3rdparty/pplx runtimecore branch to this repo's runtimecore branch.

### CODE REVIEW

File: Release/include/pplx/pplx.h [hunk 1]
```diff
@@ -133,7 +133,7 @@ class _TaskCollectionImpl
 public:
     typedef _TaskProcHandle _TaskProcHandle_t;
 
-    _TaskCollectionImpl(scheduler_ptr _PScheduler) : _M_pScheduler(_PScheduler) {}
+    _TaskCollectionImpl(scheduler_ptr _PScheduler) : _M_pScheduler(_PScheduler), _M_scheduled_task(nullptr) {}
```
Comment (dkalowsk) [inline Release/include/pplx/pplx.h]: would it make sense to just set this down on line 201 instead?  Just thinking of minimizing the changes for future upgrades.

File: Release/include/pplx/pplxtasks.h [hunk 2]
```diff
@@ -244,10 +244,7 @@ class task<void>;
 /// <seealso cref="cancellation_token Class"/>
 /// <seealso cref="cancel_current_task Function"/>
 /**/
-inline bool _pplx_cdecl is_task_cancellation_requested()
-{
-    return ::pplx::details::_TaskCollection_t::_Is_cancellation_requested();
-}
+bool _pplx_cdecl is_task_cancellation_requested();
```
Comment (dkalowsk) [inline Release/include/pplx/pplxtasks.h:247]: wow, thanks for this effort!

File: Release/include/pplx/pplxtasks.h [hunk 3]
```diff
@@ -1866,7 +1866,7 @@ class task_completion_event
         {
             ::pplx::extensibility::scoped_critical_section_t _LockHolder(_M_Impl->_M_taskListCritSec);
 
-            if (!_IsTriggered())
+            if (!_IsTriggered() && !_M_Impl->_HasUserException())
```
Comment (dkalowsk) [inline Release/include/pplx/pplxtasks.h:1865]: is this something we should push upstream to pplx?

**Review by dkalowsk — APPROVED** [review]
One minor comment that can be easily ignored.  The rest looks really good.  Nice job!

File: Release/include/pplx/pplxtasks.h (hunk 3 above)
Comment (john4744) [reply Release/include/pplx/pplxtasks.h:1865]: Yeah, I was thinking about that... Seems like the right thing to do 😄 I'll put together a PR for it.

Comment (john4744) [reply Release/include/pplx/pplxtasks.h:1865]: Submitted: https://github.com/microsoft/cpprestsdk/pull/1146

**Review by chri7325 — APPROVED** [review]
Great job!

**Pull request merged**